koku/api/iam/view/customer.py

```diff
@@ -165,3 +187,46 @@ def retrieve(self, request, *args, **kwargs):
             }
         """
         return super().retrieve(request=request, args=args, kwargs=kwargs)
+
+    @transaction.atomic
+    def destroy(self, request, *args, **kwargs):
+            """Delete a customer.
+
+            @api {delete} /api/v1/customers/:id/ Delete a customer
+            @apiName DeleteCustomers
+            @apiGroup UCustomers
+            @apiVersion 1.0.0
+            @apiDescription Delete a customer.
+
+            @apiHeader {String} token Service Admin authorizaton token.
+            @apiHeaderExample {json} Header-Example:
+                {
+                    "Authorizaton": "Token 45138a913da44ab89532bab0352ef84b"
+                }
+
+            @apiParam {Number} id Customer unique ID.
+
+            @apiSuccessExample {json} Success-Response:
+                HTTP/1.1 204 NO CONTENT
+            """
+            customer = model.Customer.objects.filter(uuid=kwargs['uuid']).first()
```

Comment:
You could use `get_object_or_404()` here, as it may short circuit the logic if the uuid is not found and first has no data.  
https://github.com/project-koku/koku/blob/master/koku/api/iam/view/user.py#L241